pkg/sensors/program/coll.go

```diff
@@ -0,0 +1,168 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Tetragon
+
+package program
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/tetragon/pkg/logger"
+)
+
+// The idea of LoadedCollection is to keep loaded programs and maps
+// from ebpf.Collection to be used later for TestLoad* tests to verify
+// loaded programs and maps.
+//
+// We can't just iterate all load programs and maps for 2 reasons:
+//   - we could race with others loading bpf programs
+//   - we get limited name, which we usually cross
+//
+// The process of loading LoadedCollection is following:
+//
+//   - ebpf collection is loaded
+//
+//     coll := NewCollectionWithOptions
+//
+//   - copy all programs/maps from ebpf collection
+//
+//     copyLoadedCollection(lc, coll)
+//
+//   - collection is removed and only 'used' programs and maps stay loaded
+//     so we filter out unused programs/maps with
+//
+//     load.LC = filterLoadedCollection(lc)
+//
+// This was we have only programs/maps realted to our test and with
```

Comment:
Not sure I fully understand the above. How about:
> Now load.LC contains programs/maps that are related to our test, are actually loaded, and have their original long names.

Reply:
LC contains arrays of program and maps that are related to the object we loaded plus we filtered out unused program and maps that we unloaded.. all this with original long names for both programs and maps